pandas-stubs/core/strings.pyi

```diff
@@ -172,15 +176,27 @@ class StringMethods(NoNewAttributesMixin, Generic[T, _TS, _TM, _TS2]):
         self, pat: str, flags: int = ..., *, expand: Literal[True] = ...
     ) -> pd.DataFrame: ...
     @overload
-    def extract(self, pat: str, flags: int, expand: Literal[False]) -> T: ...
+    def extract(
+        self, pat: str, flags: int, expand: Literal[False]
+    ) -> Series[type[object]]: ...
     @overload
-    def extract(self, pat: str, flags: int = ..., *, expand: Literal[False]) -> T: ...
+    def extract(
+        self, pat: str, flags: int = ..., *, expand: Literal[False]
+    ) -> Series[type[object]]: ...
     def extractall(self, pat: str, flags: int = ...) -> pd.DataFrame: ...
-    def find(self, sub: str, start: int = ..., end: int | None = ...) -> T: ...
-    def rfind(self, sub: str, start: int = ..., end: int | None = ...) -> T: ...
+    def find(
+        self, sub: str, start: int = ..., end: int | None = ...
+    ) -> Series[int]: ...
```

Comment:
This won't work for `Index`:
```python
>>> ind = pd.Index(["abc", "cba", "def"])
>>> ind.str.find("a")
Index([0, 2, -1], dtype='int64')
```
So I think for `StringMethods`, you have to add an extra argument to the `Generic` parameter that would be `Series[int]` or `Index[int]` based on the return types of the methods that return the sequences of `int`

Similar to how it is done for `expand`